### DISCUSSION TITLE
Option to attach a name to a Room

### DISCUSSION THREAD

**Discussion by joente** [thingsdb/ThingsDB#392] [Ideas]
Currently, rooms can only be joined using their unique ID. To enhance usability, we propose allowing rooms to have optional, unique name within their scope. This would enable users to join rooms by name, making the process more intuitive and user-friendly.

**Comment by joente** [comment]
Function to set or unset the name:

```javascript
room().set_name(...);
// argument can be a string or explicit nil to remove the name
```

Both `on_join` and `on_leave` protocol should accept both id's and names

**Comment by stefanak-michal** [comment]
So we have to allow string[], not only int[] for join and leave? what about emit, there is a int room_id? I talk in socket methods concept.

- https://docs.thingsdb.io/v1/connect/socket/join/
- https://docs.thingsdb.io/v1/connect/socket/leave/
- https://docs.thingsdb.io/v1/connect/socket/emit/

**Reply by joente** [reply]
Support for join, leave, and emit operations by the room name can be added, but is not required. The name will not replace the id for a room, but will be added next to the id. For join/leave and emit, both the name or id can be used to find a room. 
If only the name is known, a ThingsDB query like `room(name).id();` can be used to retrieve the corresponding room ID.

**Reply by stefanak-michal** [reply]
That means there are no planned changes for socket messages?

**Reply by joente** [reply]
They will stay the same, they always return with the room Id (no changes).

**Reply by joente** [reply]
I've updated the documentation:
- https://docs.thingsdb.io/v1/connect/socket/join/
- https://docs.thingsdb.io/v1/connect/socket/leave/
- https://docs.thingsdb.io/v1/connect/socket/emit/

**Reply by joente** [reply]
Testing can be done using [v1.7.0-alpha5](https://github.com/thingsdb/ThingsDB/tree/refs/tags/v1.7.0-alpha5)
